sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/DecorrelateInnerQuerySuite.scala

```diff
@@ -333,6 +334,114 @@ class DecorrelateInnerQuerySuite extends PlanTest {
     check(innerPlan, outerPlan, correctAnswer, Seq(x <=> x, y <=> y, z <=> z))
   }
 
+  test("INTERSECT ALL in correlation path") {
+    val outerPlan = testRelation2
+    val innerPlan =
+      Intersect(
+        Filter(And(OuterReference(x) === a, c === 3),
+          testRelation),
+        Filter(And(OuterReference(y) === b, c === 6),
+          testRelation),
+        isAll = true)
+    val correctAnswer =
+      Intersect(
+        Project(Seq(a, b, c, x, y),
+          Filter(And(x === a, c === 3),
+            DomainJoin(Seq(x, y),
+              testRelation))),
+        Project(Seq(a, b, c, x, y),
+          Filter(And(y === b, c === 6),
+            DomainJoin(Seq(x.newInstance(), y.newInstance()),
+              testRelation))),
+        isAll = true
+      )
+    // Disable checkAnalysis because otherwise duplicate attributes hit
```

Comment:
Hmm why do we have this issue for set operations? Can we update the test to bypass this issue?

Reply:
It's a specific analysis check for the two sides of intersect/except having disjoin attributes:
```
case i: Intersect if !i.duplicateResolved =>
            val conflictingAttributes = i.left.outputSet.intersect(i.right.outputSet)
            i.failAnalysis(
```
I was able to modify the test to bypass it - previously I tried but got an issue with missing attributes, this time I was able to modify the test to get around that too.